pkg/operator/ceph/config/livenessprobe.go

```diff
@@ -51,3 +51,28 @@ func ConfigureLivenessProbe(daemon rookv1.KeyType, container v1.Container, healt
 
 	return container
 }
+
+func GetLivenessProbeWithDefaults(desiredProbe, defaultProbe *v1.Probe) *v1.Probe {
+	if desiredProbe.Handler.Exec == nil && desiredProbe.Handler.HTTPGet == nil && desiredProbe.Handler.TCPSocket == nil {
```

Comment:
```suggestion
	if desiredProbe.Handler == v1.Handler{} {
```

This should do it.

Reply:
yes I tried that but it shows `expected ';', found '{'syntax` in the vscode I might be doing something wrong

Reply:
Edited, it should be fine now!

Reply:
Or use `!reflect.DeepEqual(desiredProbe.Handler, (v1.Handler{}))` if you want, we do that in other places.
Reflect is probably better though...

Reply:
Actually reflect is useful if the type has underlying slices, so we might not need it, let's keep it simple with my initial solution!

Reply:
Okay, btw thanks for the knowledge

Reply:
Done.